packages/terraform-modules-public/testnet/modules/validator/startup.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+# ---- Set Up Logging ----
+
+curl -sSO https://dl.google.com/cloudagents/install-logging-agent.sh
+bash install-logging-agent.sh
+
+# ---- Set Up Persistent Disk ----
+
+# gives a path similar to `/dev/sdb`
+DISK_PATH=`readlink -f /dev/disk/by-id/google-${attached_disk_name}`
+DATA_DIR=/root/.celo
+
+echo "Setting up persistent disk ${attached_disk_name} at $DISK_PATH..."
+
+DISK_FORMAT=ext4
+CURRENT_DISK_FORMAT=`lsblk -i -n -o fstype $DISK_PATH`
+
+echo "Checking if disk $DISK_PATH format $CURRENT_DISK_FORMAT matches desired $DISK_FORMAT..."
+
+# If the disk has already been formatted previously (this will happen
+# if this instance has been recreated with the same disk), we skip formatting
+if [[ $CURRENT_DISK_FORMAT == $DISK_FORMAT ]]; then
+  echo "Disk $DISK_PATH is correctly formatted as $DISK_FORMAT"
+else
+  echo "Disk $DISK_PATH is not formatted correctly, formatting as $DISK_FORMAT..."
+  mkfs.ext4 -m 0 -F -E lazy_itable_init=0,lazy_journal_init=0,discard $DISK_PATH
+fi
+
+echo "Mounting $DISK_PATH onto $DATA_DIR"
+mkdir -p $DATA_DIR
+mount -o discard,defaults $DISK_PATH $DATA_DIR
```

Comment:
do you need to update the fstab (so the fs mounts on boot)?

Reply:
e.g.,

```
echo "Mounting $DISK_PATH onto $DATA_DIR"
mkdir -p $DATA_DIR
DISK_UUID=`blkid $DISK_PATH | cut -d \" -f 2`
echo "UUID=${DISK_UUID}     $DATA_DIR   auto    discard,defaults    0    0" >> /etc/fstab
mount $DISK_PATH
```

Reply:
Yes, also makes sense. Really the module is not tested/thought to restart the instance but we can add support to it, so if someone manages the instances manually this will help.